romancal/dq_init/dq_init_step.py

```diff
@@ -63,6 +64,14 @@ def process(self, input):
         else:
             init_model = input_model
 
+        # guide window range information
+        x_start = input_model.meta.guidestar.gw_window_xstart
+        x_end = input_model.meta.guidestar.gw_window_xsize + x_start
+        # set pixeldq array to GW_AFFECTED_DATA (2**4) for the given range
+        init_model.pixeldq[int(x_start):int(x_end),:] = dqflags.pixel['GW_AFFECTED_DATA']
```

Comment:
I wonder why the casting to integer here? I wanted to check if `xstart` and `ystart` are defined as `float` but as mentioned above I don't see them in the schema. In any case, assuming they are floats, just using `int` is not going to be sufficient. We define the center of the pixel to be integer and the first pixel goes from -0.5 to +0.5. Because of this convention just using `int` does not always give the correct pixel. In cases like this we should use `from gwcs.utils import _toindex` which does this correctly.

Reply:
The gw_window_xstart come down in telemetry as in unsigned integer and should be stored as that in the schema. They are currently floats in the schema so when doing the math the pixel locations are floats but need to be an integer when updating the array.